packages/mdc-slider/foundation.js

```diff
@@ -0,0 +1,248 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ *you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {attributes} from './constants';
+import MDCSliderAdapter from './adapter';
+
+import MDCFoundation from '@material/base/foundation';
+
+/** @enum {string} */
+const MOVE_EVENT_MAP = {
+  'mousedown': 'mousemove',
+  'touchstart': 'touchmove',
+  'pointerdown': 'pointermove',
+};
+
+const DOWN_EVENTS = ['mousedown', 'pointerdown', 'touchstart'];
+const UP_EVENTS = ['mouseup', 'pointerup', 'touchend'];
+
+/**
+ * @extends {MDCFoundation<!MDCSliderAdapter>}
+ */
+class MDCSliderFoundation extends MDCFoundation {
+  /** @return {!MDCSliderAdapter} */
```

Comment:
The foundation should always expose the constants for someone wrapping the foundation in another framework, like:
```
  /** @return enum {string} */
  static get selectors() {
    return selectors;
  }
```
Remember to add a test for this too!

Reply:
+1